src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -81,13 +81,12 @@ public static DataGroup convertFitsToDataGroup(String fits_filename,
                                                          int table_idx) throws FitsException, IOException {
 
 
-        Fits fits= null;
-        try {
+        FitsFactory.useThreadLocalSettings(true);
+        try (Fits fits = new Fits(fits_filename)) {
             // disable long string for HeaderCard creation while collecting table with table_idx from StarTableFactory to work around
             // the exception error sent from nom.tam.fits.
```

Comment:
I think `useThreadLocalSettings(true)` was added as a workaround for StarTableFactory.  If the updated Starlink is no longer using Fits, we can safely remove this workaround.  It's not foolproof anyways(multi-threads race condition).